upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1399,15 +1398,21 @@ func (n *nodeUpConfigBuilder) BuildConfig(ig *kops.InstanceGroup, apiserverAddit
 	return config, bootConfig, nil
 }
 
-func getTasksCertificate(caTasks map[string]*fitasks.Keypair, name string, config *nodeup.Config, includeKeypairID bool) error {
-	cas, err := fi.ResourceAsString(caTasks[name].Certificates())
+func loadCertificate(keysets map[string]*fi.Keyset, name string, config *nodeup.Config, includeKeypairID bool) error {
+	keyset := keysets[name]
+	if keyset == nil {
+		return fmt.Errorf("key %q not found", name)
+	}
+	certificates, err := keyset.ToCertificateBytes()
 	if err != nil {
-		// CA task may not have run yet; we'll retry
-		return fmt.Errorf("failed to read %s certificates: %w", name, err)
+		return fmt.Errorf("error getting certificate from keyset %q: %w", name, err)
```

Comment:
My understanding is that the word "error" is usually used before printing an `error`'s string, so I try to avoid using the word in an error's string.

Reply:
Ah, I like that rule.  This wasn't a deliberate rephrase, I think I came over a few iterations, so I'll just revert this